microsoft/testsuites/vm_extensions/cvt.py

```diff
@@ -0,0 +1,284 @@
+) -> ExecutableResult:
+    timer = create_timer()
+    script: CustomScript = node.tools[cvt_script]
+    result = script.run(parameters=test_dir, timeout=19800, sudo=True)
+    log.info(f"Script run with param {test_dir} finished within {timer}")
+    return result
+
+
+def _copy_cvt_logs(
+    log: Logger,
+    log_path: Path,
+    node: Node,
+    test_dir: Path,
+) -> None:
+    find_tool = node.tools[Find]
+    file_list = find_tool.find_files(
+        test_dir,
+        name_pattern="*.log",
+        file_type="f",
+        sudo=True,
+        ignore_not_exist=True,
+    )
+
+    file_list += find_tool.find_files(
+        test_dir,
+        name_pattern="*.txt",
+        file_type="f",
+        sudo=True,
+        ignore_not_exist=True,
+    )
+
+    for file in file_list:
+        log.info(f"Copying file {file} to {log_path}")
+        try:
+            file_name = file.split("/")[-1]
+            node.shell.copy_back(
+                node.get_pure_path(file),
+                log_path / f"{file_name}",
+            )
+        except FileNotFoundError:
+            log.error(f"File {file} doesn't exist.")
+
+
+def _run_cvt_tests(
+    log: Logger,
+    node: Node,
+    log_path: Path,
+    variables: Dict[str, Any],
+    os: str,
+    cvt_script: CustomScriptBuilder,
+) -> Optional[int]:
+    cvt_bin = "indskflt_ct"
+    container_sas_uri = variables.get("cvtbinaries_sasuri", "")
+    if not container_sas_uri:
+        log.error("sas uri for cvt binary is empty.")
+        return 1
+    cvt_binary_sas_uri = container_sas_uri.replace(
+        "?", "/cvtbinaries/indskflt_ct_" + os + "?"
+    )
+
+    cvt_root_dir = str(node.working_path) + "/LisaTest/"
+    cvt_download_dir = cvt_root_dir + "cvt_files/"
+
+    mkdir = node.tools[Mkdir]
+    wget = node.tools[Wget]
+
+    mkdir.create_directory(cvt_download_dir, sudo=True)
+    download_path = wget.get(
+        url=f"{cvt_binary_sas_uri}",
+        filename=cvt_bin,
+        file_path=cvt_download_dir,
+        sudo=True,
+    )
+
+    cvt_md5sum = node.execute(f"md5sum {download_path}", shell=True, sudo=True)
+    log.info(f"md5sum '{download_path}' : '{cvt_md5sum}'")
+
+    result = _run_script(
+        node=node, log=log, test_dir=cvt_download_dir, cvt_script=cvt_script
+    )
+    _copy_cvt_logs(
+        node=node,
+        log=log,
+        test_dir=Path(node.working_path),
+        log_path=log_path,
+    )
+    return result.exit_code
+
+
+@TestSuiteMetadata(
+    area="cvt",
+    category="functional",
+    description="""
+    This test is used to validate the functionality of ASR driver.
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class CVTTest(TestSuite):
+    TIMEOUT = 21600
+
+    @TestCaseMetadata(
+        description="""
+        this test validate the functionality of ASR driver by verifying
+        integrity of a source disk with respect to a target disk
+        """,
+        priority=1,
+        use_new_environment=True,
```

Comment:
Why is it need a new environment? If there is no special reason, it's better to reuse existing environment, so the test run can be faster by reducing provisioning.